### ISSUE TITLE
C extensions are not being built inside tox builds

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8724]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8724|
|Type|enhancement|
|Created|2016-08-04 21:29:56Z|
|Branch|https://github.com/twisted/twisted/tree/8724-rodrigc-tox-ext|

C extensions like raiser.c
or the IOCP reactor on Windows are not being built when
testing things with tox

<details><summary>Searchable metadata</summary>

```
trac-id__8724 8724
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__None None
branch__8724_rodrigc_tox_ext 8724-rodrigc-tox-ext
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1470346196617161 1470346196617161
changetime__1470830901176709 1470830901176709
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

Are you sure? It's working for me:

```
red ~/c/twgit ((c5da63e...))> cd build/py27-nomodules/
red ~/c/t/b/py27-nomodules ((c5da63e...))> bin/python
Python 2.7.10 (default, Oct 23 2015, 19:19:21)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import twisted.test.raiser
>>> twisted.test.raiser
<module 'twisted.test.raiser' from '/Users/red/code/twgit/build/py27-nomodules/lib/python2.7/site-packages/twisted/test/raiser.so'>
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl set status to `closed`|
|-|-|

I think this is a misdiagnosed [#8725](https://github.com/twisted/twisted/issues/8725) -- the IOCP tests are being run on AppVeyor, it's just that coverage isn't combining correctly. See: https://codecov.io/gh/twisted/twisted/tree/3e488bca70159bd8d7f5a62dc55e0ab9879f229b/build/py27-coverage-windows

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set status to `reopened`|
|-|-|

Yes, there are some problems with how things are configured, and how things
are built.  This is more visible when compiling and running the tests with Python 3 on Windows.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

See: https://github.com/twisted/twisted/pull/463

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @rodrigc<br>@glyph set status to `new`|
|-|-|

There's a license concern with the CC0 batch file that blocks this until we can hear back from a lawyer, but otherwise I'm happy to approve it.  Longer term we should probably get rid of that file anyway though, and replace it with a Python script, ideally something that others can `pip install`, since this is a clearly a problem that more than one Python project has had.  If it takes a long time to hear back from a lawyer then you can do that preemptively and re-submit ;).

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset c462cc1140cec9739676fe314fb940bc9a919f11
```
#!CommitTicketReference repository="" revision="c462cc1140cec9739676fe314fb940bc9a919f11"
Merge pull request #463 from twisted/8724-rodrigc-tox-ext

Author: rodrigc
Reviewer: glyph
Fixes: #8724

Fix building of C extensions, especially visible on Windows
```
